tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/Refiner.scala

```diff
@@ -149,12 +149,10 @@ private object Refiner {
 
   private class LinkedClassInfoCache {
     private var cacheUsed: Boolean = false
-    private var classVersion: Option[String] = None
-    private var staticMethodsVersions: List[(String, Option[String])] = Nil
-    private var memberMethodsVersions: List[(String, Option[String])] = Nil
-    private var abstractMethodsVersions: List[(String, Option[String])] = Nil
-    private var exportedMembersVersions: List[(String, Option[String])] = Nil
-    private var topLevelExportsVersions: List[(String, Option[String])] = Nil
+    private val staticMethodsInfoCaches = LinkedMembersInfosCache()
+    private val memberMethodsInfoCaches = LinkedMembersInfosCache()
+    private val abstractMethodsInfoCaches = LinkedMembersInfosCache()
+    private val exportedMembersInfoCaches = LinkedMembersInfosCache()
     private var info: Infos.ClassInfo = _
```

Comment:
In case `loadInfo()` is called twice during the same run (see also `if (!cacheUsed)` in `update`). Currently, the analyzer and the IR checker should never call it twice for the same class (even together), but I'd rather not endanger performance in subsequent changes.